apps/meteor/app/e2e/server/functions/provideUsersSuggestedGroupKeys.ts

```diff
@@ -0,0 +1,56 @@
+import type { IRoom, IUser } from '@rocket.chat/core-typings';
+import { Rooms, Subscriptions } from '@rocket.chat/models';
+import { Meteor } from 'meteor/meteor';
+
+import { isTruthy } from '../../../../lib/isTruthy';
+import { canAccessRoomIdAsync } from '../../../authorization/server/functions/canAccessRoom';
+
+export const provideUsersSuggestedGroupKeys = async (
+	usersSuggestedGroupKeys: Record<IRoom['_id'], { _id: IUser['_id']; key: string }[]>,
+) => {
+	const userId = Meteor.userId();
+
+	if (!userId) {
+		throw new Meteor.Error('error-invalid-user', 'Invalid user');
+	}
+
+	const roomIds = Object.keys(usersSuggestedGroupKeys);
+
+	if (!roomIds) {
+		return;
+	}
+
+	for await (const roomId of roomIds) {
+		if (!(await canAccessRoomIdAsync(roomId, userId))) {
+			throw new Meteor.Error('error-invalid-room', 'Invalid room');
+		}
+	}
+
+	const completedRoomIds = (
+		await Promise.all(
+			roomIds.map(async (roomId) => {
+				const users = usersSuggestedGroupKeys[roomId];
+
+				const subscription = await Subscriptions.findOneByRoomIdAndUserId(roomId, userId);
+
+				if (!subscription) {
+					return;
+				}
```

Comment:
We don't need to check this, we already checked the `canAccessRoomIdAsync`. We are good.